addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -145,6 +145,15 @@ def _check_unique_sequence_number(self):
         vendor = self.filtered(lambda x: x.is_purchase_document() and x.l10n_latam_use_documents)
         return super(AccountMove, self - vendor)._check_unique_sequence_number()
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_forbid_parts_of_chain(self):
+        """ Moves with a sequence number can only be deleted if they are the last element of a chain of sequence.
+        If they are not, deleting them would create a gap. If the user really wants to do this, he still can
+        explicitly empty the 'name' field of the move; but we discourage that practice.
+        """
+        vendor = self.filtered(lambda x: x.is_purchase_document() and x.l10n_latam_use_documents)
```

Comment:
Should not this be everywhere we have a manual document number then?  It would not be bad to generalize getting a manual document number for that.

Reply:
@zaoral Hy Kathy, I think are we including moves with manual document numbers really, is that right?

Reply:
Hi, @jco-odoo I made a change to consider the manual document number.